src/Game/AccountManagement.java

```diff
@@ -356,8 +1034,153 @@ public static void account_panels_input_hook(int n1, int mouseY, int a, int n3,
     }
   }
+  
+  public static void recovery_questions_input(int n1, int mouseY, int n3, int mouseX) {
+	  try {
+		  if (customQuestionEntry != -1) {
+			  if (Client.pm_text.length() > 0) {
+				  Client.controlRecoveryText[customQuestionEntry] = Client.pm_text;
+				  Panel.setControlText(
+				            Client.panelRecoveryQuestions,Client.controlRecoveryIns[customQuestionEntry], customQuestionEntry + 1 + ": " + Client.controlRecoveryText[customQuestionEntry]);
+				  Panel.setControlText(
+				            Client.panelRecoveryQuestions,Client.controlAnswerInput[customQuestionEntry], "");
+		            customQuestionEntry = -1;
+			  }
+		  } else {
+			  Panel.handleMouse(Client.panelRecoveryQuestions, n1, mouseY, n3, mouseX);
+			  
+			  int cursorIndex;
+				for (int questionIndex = 0; questionIndex < 5; ++questionIndex) {
+					if (Panel.isSelected(Client.panelRecoveryQuestions, Client.controlQuestion[questionIndex])) {
+						boolean hasDistinctQuestions = false;
+
+						while (!hasDistinctQuestions) {
+							recoveryIndices[questionIndex] = (recoveryIndices[questionIndex] + 1) % recoveryQuestions.length;
+							hasDistinctQuestions = true;
+
+							for (cursorIndex = 0; cursorIndex < 5; ++cursorIndex) {
+								if (cursorIndex != questionIndex && recoveryIndices[cursorIndex] == recoveryIndices[questionIndex]) {
+									hasDistinctQuestions = false;
+								}
+							}
+						}
+						
+						Client.controlRecoveryText[questionIndex] = recoveryQuestions[recoveryIndices[questionIndex]];
+			            Panel.setControlText(Client.panelRecoveryQuestions,Client.controlRecoveryIns[questionIndex], questionIndex + 1 + ": " + recoveryQuestions[recoveryIndices[questionIndex]]);
+			            Panel.setControlText(Client.panelRecoveryQuestions,Client.controlAnswerInput[questionIndex], "");
+					}
+				}
+				
+				for (int i = 0; i < 5; ++i) {
+					if (Panel.isSelected(Client.panelRecoveryQuestions,Client.controlCustomQuestion[i])) {
+						customQuestionEntry = i;
+		                Client.pm_enteredText = "";
+		                Client.pm_text = "";
+					}
+				}
+				
+				if (Panel.isSelected(Client.panelRecoveryQuestions,Client.chooseFinishSetRecoveryButton)) {
+					cursorIndex = 0;
+					
+					while (true) {
+						if (cursorIndex >= 5) {
+							for (int outerIdx = 0; outerIdx < 5; ++outerIdx) {
+								String checkedAnswer = Panel.getControlText(Client.panelRecoveryQuestions, Client.controlAnswerInput[outerIdx]);
+
+								for (int innerIdx = 0; innerIdx < outerIdx; ++innerIdx) {
+									String questionAnswer = Panel.getControlText(Client.panelRecoveryQuestions, Client.controlAnswerInput[innerIdx]);
+									if (checkedAnswer.equalsIgnoreCase(questionAnswer)) {
+										Panel.setControlText(Client.panelRecoveryQuestions,Client.controlRecoveryQuestions, "@yel@Each question must have a different answer");
+										return;
+									}
+								}
+							}
+
+							sendRecoveryQuestions();

```

Comment:
might be nice to leave a comment here that this generates & sends RSC175 opcode 208, so it doesn't blend into the rest of the garbage as easily

Reply:
fixed